statsmodels/base/dimensions.py

```diff
@@ -0,0 +1,232 @@
+	"""
+	if assume_const:
+		k_const = 1.0
+	else:
+		k_const = inst.k_constant
+
+	if assert_const:
+		assert inst.k_constant == 1, inst.k_constant
+	return k_const
+
+
+# Inherited by Model and Results
+class NEQsMixin(object):
+	
+	@cache_readonly
+	def neqs(self):
+		try:
+			# Note: Recall that MLEModel and sub-classes have endog transposed
+			# TODO: does the above apply to any of DiscreteModel?
+			return self.endog.shape[1] # FIXME: does the MLEModel note above mean that this will be wrong?
+		except (AttributeError, IndexError):
+			# Note: np.ndim(None) is 0
+			return np.ndim(self.endog)
+
+	k_endog = neqs # TODO: Is this a misnomer for MultinomialModel, MNLogit?
+
+
+# Inherited by VARResults, Model, and Results
+class NobsMixin(object):
+	@cache_readonly
+	def _nobs_total(self):
+		"""
+		AR, ARIMA, VAR, ... models have `nobs` that is not always equal
+		to `len(endog)`.  `_nobs_total` gives the *total* observations,
+		including any that are dropped or "burned" like in these special
+		cases.
+		"""
+		try:
+			# regime_switching
+			return len(self.orig_endog) # TODO: should we use self.data.orig_endog?
+		except AttributeError:
+			return len(self.endog)
+			# FIXME: This may break in models where self.endog is messed with.
+
+	@cache_readonly
+	def nobs(self):
+		# Note: we could use endog.shape[0], but in at least
+		# one test case self.endog is a list
+		return len(self.endog)
+		# Notes from the GEE Case:
+		# v1 = sum([len(y) for y in self.endog_li])
+		# v2 = sum([len(y) for y in self.group_indices.values()])
+		# v3 = len(self.endog)
+		# assert v1 == v2, (v1, v2)
+		# assert v3 == v1, (v3, v1)
+
+
+class KarNobsMixin(object):
+	@property
+	def nobs(self):
+		# In markov_switching and markov_autoregression, _nobs_total comes
+		# from len(self.orig_endog).  markov_switching does not have a
+		# k_ar attribute, but markov_autoregression (which subclasses markov_swiching)
+		# does.
+		#
+		# In linear_model, this is equivalent to self.wexog.shape[0].
+		# For OLS, WLS, and GLS this is the same as _nobs_total, i.e.
+		# len(self.endog).  However for GLSAR the `whiten` method
+		# returns a `self.wexog` which is shorter than `self.endog` by `k_ar`,
+		# so we subtract that term to get `nobs`.
+		return self._nobs_total - getattr(self, 'k_ar', 0)
+
+
+class CssKarNobsMixin(object):
+	@property
+	def nobs(self):
+		n_totobs = self._nobs_total
+		if self.method in ["cmle", "css", "ols"]:
+			nobs = (n_totobs - self.k_ar)
+		else:
+			nobs = n_totobs
+		# ARModel used to set nobs = n_totobs iff method == "mle" and
+		# otherwise raise NotImplementedError
+		return float(nobs)
+
+
+class WNobsMixin(object):
+	"""
+
+	This mixin is specific to GLM Model, defines a `wnobs` property in
+	terms of the model's `freq_weights` attribute.
+
+	"""
+
+	@property
+	def wnobs(self):
+		if (self.freq_weights is not None) and (self.freq_weights.shape[0] == self.endog.shape[0]):
+			wnobs = bn.nansum(self.freq_weights) # self.freq_weights.sum()
```

Comment:
use np.nansum, bottleneck is not a statsmodels dependency
the test failures on travis are uninformative about this PR, because they never get over importing this module